sqrl-openai/src/main/java/com/datasqrl/openai/FunctionExecutor.java

```diff
@@ -4,32 +4,44 @@
 import org.apache.flink.table.functions.FunctionContext;
 
 import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
 import static com.datasqrl.openai.RetryUtil.executeWithRetry;
 
 public class FunctionExecutor {
 
     private final FunctionMetricTracker metricTracker;
+    private final ExecutorService executorService;
 
     public FunctionExecutor(FunctionContext context, String functionName) {
         this.metricTracker = new FunctionMetricTracker(context, functionName);
+        this.executorService = Executors.newFixedThreadPool(10);
     }
 
-    public <T> T execute(Callable<T> task) {
-        metricTracker.increaseCallCount();
-
-        final long start = System.nanoTime();
-
-        final T ret = executeWithRetry(
-                task,
-                metricTracker::increaseErrorCount,
-                metricTracker::increaseRetryCount
-        );
-
-        final long elapsedTime = System.nanoTime() - start;
-        metricTracker.recordLatency(TimeUnit.NANOSECONDS.toMillis(elapsedTime));
-
-        return ret;
+    public <T> CompletableFuture<T> executeAsync(Callable<T> task) {
+        final CompletableFuture<T> future = new CompletableFuture<>();
+        executorService.submit(() -> {
+            try {
+                metricTracker.increaseCallCount();
+                final long start = System.nanoTime();
+
+                T result = executeWithRetry(
```

Comment:
Flink async functions have retry build-in now:
https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/config/#table-exec-async-scalar-retry-strategy

Please remove the retry logic and just keep the metrics piece.